app/files.js

```diff
@@ -312,6 +375,42 @@ const roomArchived = {
   }
 }
 
+const shareRoles = {
```

Comment:
> I'm not sure why we need a hidden trigger. Wouldn't it be better to use it as part of another trigger instead?

The hidden trigger allows us to get the room type from the id of the room the user has chosen and return the roles allowed for that room. Since we can't know in advance which room the user will choose, we can't generate a correct list of roles. In our case, if the room is a "Public Room", the available roles for it are Viewer and Editor. For a "Custom Room", Room Admin and Power User.